client/state/plugins/installed/types.ts

```diff
@@ -0,0 +1,65 @@
+import type { MomentInput } from 'moment';
+
+// The installed plugins found in the Redux store
+export type InstalledPlugins = {
+	[ key: string ]: InstalledPluginData[];
+};
+
+export type InstalledPluginData = {
+	active: boolean;
+	author: string;
+	author_url: string;
+	autoupdate: boolean;
+	description: string;
+	id: string;
+	name: string;
+	network: boolean;
+	plugin_url: string;
+	slug: string;
+	uninstallable: boolean;
+	update?: PluginUpdate;
+	version: string;
+};
+
+// This is the plugin as it is exposed by the selectors
+export type Plugin = {
+	action_links?: { Settings?: string; name?: string };
+	fetched: boolean;
+	icon: string;
+	id: string;
+	last_updated: MomentInput;
+	name: string;
+	sites: PluginSites;
+	slug: string;
+	statusRecentlyChanged?: boolean;
+	update?: PluginUpdate;
+	wporg?: boolean;
+};
+
+export type PluginSites = {
+	[ key: string ]: {
+		active: boolean;
+		autoupdate: boolean;
+		update?: PluginUpdate;
+		version: string;
```

Comment:
It could be done that way technically but I'm not sure that conceptually it works. `InstalledPluginData` is intended to reflect what is in the Redux store, which in turn is what comes from the server. `PluginSites` is the shape of objects computed from that server data. 

So I don't see `InstalledPluginData` as being composed out of any sub-types, it is simply what comes from the server. Even if I made this type a new type I wouldn't want to use it in `InstalledPluginData` as I find it easiest if the data which comes from a server is clearly represented as its own type, which makes it easiest for future developers to update.

Reply:
I don't have particularly strong feelings about it, it just feels like those types are a bit complex and still have some repetition, so it might make sense to try improving them in the future. I do agree that the data returned by the server and the data originally on the client don't always correspond and are historically a bit messy.